### PR TITLE
Python 3 support

### LINKED ISSUE #44: Support Python 3
We should be mostly compatible, but when running `pipenv install terraformpy` with Python 3 available the install fails since this package is not marked as universal and supporting 3+.

### LINKED ISSUE #52: Readme example fails to run
Toying around with `terraformpy`, I attempted to run the example in the `README.md` file, but I was not able to completed the process:

`main.tf.py`:
```python
from terraformpy import Provider, Resource

Provider("aws", profile="default", region="us-east-1")

Resource("aws_instance", "example", ami="ami-2757f631", instance_type="t2.micro")
```

output:
```bash
$ terraformpy
terraformpy - Processing: main.tf.py
Traceback (most recent call last):
  File "/Users/remygreinhofer/remy-project/labs/terraformpy-test-project/venv/bin/terraformpy", line 11, in <module>
    load_entry_point('terraformpy', 'console_scripts', 'terraformpy')()
  File "/Users/remygreinhofer/remy-project/labs/terraformpy-test-project/vendor/terraformpy/src/terraformpy/cli.py", line 42, in main
    imp.load_source(filename[:-6], filename)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "main.tf.py", line 3, in <module>
    Provider("aws", profile="default", region="us-east-1")
  File "/Users/remygreinhofer/remy-project/labs/terraformpy-test-project/vendor/terraformpy/src/terraformpy/objects.py", line 89, in __new__
    inst = super(TFObject, cls).__new__(cls, *args, **kwargs)
TypeError: object.__new__() takes exactly one argument (the type to instantiate)
```

This happened using `terraformpy 1.1.2`.

### PR DESCRIPTION
**Pull request by borgstrom**
This PR adds support for Python 3.5, 3.6, 3.7 & 3.8.  CI now runs the test suite against these versions as well as 2.7 and the wheel published is marked as universal.

Closes #52 
Closes #44

### CODE REVIEW

File: src/terraformpy/objects.py [hunk 1]
```diff
@@ -42,15 +42,40 @@ class DuplicateKey(str):
...
+        inst.hash = DuplicateKey.next_hash_per_str[key]
+        DuplicateKey.next_hash_per_str[key] = inst.hash + 1
+
+        return inst
 
     def __hash__(self):
         return self.hash
 
+    def __eq__(self, other):
+        return self.__class__ == other.__class__ and self.hash == other.hash
```
Comment (borgstrom) [inline src/terraformpy/objects.py]: Per [the Python 3 data model docs on `__hash__`](https://docs.python.org/3/reference/datamodel.html#object.__hash__):

> If a class does not define an `__eq__()` method it should not define a `__hash__()` operation either

Without `__eq__` and the other rich comparison methods the py3 tests were failing with:

```
______________________________ test_duplicate_key ______________________________

    def test_duplicate_key():
        # ordering should ensure it's always <firstCreatedKey>, <secondCreatedKey>
        # so let's ensure that works, even if say the value names are backwards-sorted
        key2 = DuplicateKey("mysql")
        key1 = DuplicateKey("mysql")
>       encoded = json.dumps({key1: {"user": "wyatt1"}, key2: {"user": "wyatt2"}}, sort_keys=True)

tests/test_objects.py:183: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py:238: in dumps
    **kw).encode(obj)
/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py:199: in encode
    chunks = self.iterencode(o, _one_shot=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.encoder.JSONEncoder object at 0x10974b150>, o = {'mysql': {'user': 'wyatt1'}, 'mysql': {'user': 'wyatt2'}}, _one_shot = True

    def iterencode(self, o, _one_shot=False):

<snip...>

>       return _iterencode(o, 0)
E       TypeError: '<' not supported between instances of 'dict' and 'dict'
```

**Comment by ss-remygreinhofer** [comment]
Questions/suggestions;
* Do you need to keep the Python 2 version? It seems to add more complexity for no good reason.
* Have you considered using [multidict](https://github.com/aio-libs/multidict) instead of re-implementing `DuplicateKey` yourself? It looks like it would fit exactly your needs in terms of preserving insertion order and allowing for having multiple values for the same key.

**Comment by borgstrom** [comment]
> Do you need to keep the Python 2 version? It seems to add more complexity for no good reason.

Yes, we do.  Our internal Terraform implementation is still Python 2, and while EOL is right around the corner we won't be able to move to Python 3 before this PR is merged. 😅 

We _could_ simplify things (like `super()` without args) if we didn't have Python 2 support, but I don't think dropping it would simplify the code very much.

> Have you considered using multidict instead of re-implementing DuplicateKey yourself?

We had not, but we can't use it because it is not JSON encodable:

```
    def test_multidict():
        from multidict import MultiDict
        out = MultiDict()
        out["mysql"] = {"user": "wyatt2"}
        out["mysql"] = {"user": "wyatt1"}
>       encoded = json.dumps(out, sort_keys=True)

tests/test_objects.py:193: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py:231: in dumps
    return _default_encoder.encode(obj)
/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py:199: in encode
    chunks = self.iterencode(o, _one_shot=True)
/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py:257: in iterencode
    return _iterencode(o, 0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.encoder.JSONEncoder object at 0x1092e8990>, o = <MultiDict('mysql': {'user': 'wyatt1'})>

    def default(self, o):
        """Implement this method in a subclass such that it returns
        a serializable object for ``o``, or calls the base implementation
        (to raise a ``TypeError``).
    
        For example, to support arbitrary iterators, you could
        implement default like this::
    
            def default(self, o):
                try:
                    iterable = iter(o)
                except TypeError:
                    pass
                else:
                    return list(iterable)
                # Let the base class default method raise the TypeError
                return JSONEncoder.default(self, o)
    
        """
>       raise TypeError(f'Object of type {o.__class__.__name__} '
                        f'is not JSON serializable')
E       TypeError: Object of type MultiDict is not JSON serializable
```

**Comment by ss-remygreinhofer** [comment]
I see. It will work with a custom encoder, like what werkzeug did: https://github.com/pallets/werkzeug/blob/e7ba08f209477cb453f15113f9a4d527a6e81bfe/src/werkzeug/datastructures.py#L594

**Comment by borgstrom** [comment]
> I see. It will work with a custom encoder, like what werkzeug did: https://github.com/pallets/werkzeug/blob/e7ba08f209477cb453f15113f9a4d527a6e81bfe/src/werkzeug/datastructures.py#L594

That won't work either because it's converted back to a native `dict` and the keys are not handled in any special way, so the multiple keys would be clobbered again since they're just native strings.  Here's what the werkzeug implementation looks like:

```
    def test_multidict():
        import six
        from multidict import MultiDict
        out = MultiDict()
        out["mysql"] = {"user": "wyatt2"}
        out["mysql"] = {"user": "wyatt1"}
        encoded = json.dumps(dict(six.iteritems(out)), sort_keys=True)
        desired = '{"mysql": {"user": "wyatt2"}, "mysql": {"user": "wyatt1"}}'
>       assert encoded == desired
E       assert '{"mysql": {"...": "wyatt1"}}' == '{"mysql": {"...": "wyatt1"}}'
E         - {"mysql": {"user": "wyatt1"}}
E         + {"mysql": {"user": "wyatt2"}, "mysql": {"user": "wyatt1"}}
```

The whole point is that the JSON representation of HCL supports multiple keys with the same value (which technically isn't against the JSON spec) so we need our JSON representation to have multiple keys, which is what `DuplicateKey` does because **it is a native string** (i.e. it is a subclass of `str`) so it works for both `dict` and JSON encoded form.

**Pull request merged**